dask/array/creation.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import absolute_import
+from itertools import count
+
+import numpy as np
+from toolz import curry
+
+from .core import Array
+
+linspace_names = ('linspace-%d' % i for i in count(1))
+arange_names = ('arange-%d' % i for i in count(1))
+
+
+def _get_blocksizes(num, blocksize):
+    # compute blockdims
+    remainder = (num % blocksize,)
+    if remainder == (0,):
+        remainder = tuple()
+    blocksizes = ((blocksize,) * int(num // blocksize)) + remainder
+    return blocksizes
+
+
+def linspace(start, stop, num=50, blocksize=None, dtype=None):
+    """
+    Return `num` evenly spaced values over the closed interval [`start`,
+    `stop`].
+
+    TODO: implement the `endpoint`, `restep`, and `dtype` keyword args
+
+    Parameters
+    ----------
+    start : scalar
+        The starting value of the sequence.
+    stop : scalar
+        The last value of the sequence.
+    blocksize :  int
+        The number of samples on each block. Note that the last block will have
+        fewer samples if `num % blocksize != 0`
+    num : int, optional
+        Number of samples to in the returned dask array, including the
+        endpoints.
+
+    Returns
+    -------
+    samples : dask array
+
+    """
+    num = int(num)
+
+    if blocksize is None:
+        raise ValueError("Must supply a blocksize= keyword argument")
+
+    blocksizes = _get_blocksizes(num, blocksize)
+
+    range_ = stop - start
+
+    space = float(range_) / (num - 1)
+
+    name = next(linspace_names)
+
+    dsk = {}
+    blockstart = start
+
+    for i, bs in enumerate(blocksizes):
+        blockstop = blockstart + ((bs - 1) * space)
+        task = (curry(np.linspace, blockstart, blockstop, num=bs,
+                      dtype=dtype),)
+        blockstart = blockstart + (space * bs)
+        dsk[(name, i)] = task
```

Comment:
I don't think it is avoidable here... Unless I just past in two `None`s between `num` and `dtype`. But that seemed kind of hackish.